### ISSUE TITLE
Recall and Precision of binary classifciation not the same as manual calculated

### ISSUE THREAD

**Issue by tuxmania87** [tensorflow/tensorflow#39836]
i have a Keras Sequential Network and use the validation_data attribute to get an idea of the validation straight away. 

However when the model is trained and I predict on the validation set I dont get as high scores as keras is telling me through training.

This is the model and training:

```
def getKerasModel(ndim):

    model = Sequential()
    model.add(Dense(100, activation='relu', input_shape=(ndim,)))
    model.add(Dense(40, activation='relu'))
    model.add(Dense(1, activation='sigmoid'))
    model.compile(optimizer="adam", loss="binary_crossentropy", metrics=[keras.metrics.Precision(), keras.metrics.Recall()])
    return model
```
Then I train it:
```
from keras.wrappers.scikit_learn import KerasClassifier
from keras.utils import to_categorical

# pipeline is doing scales and one hot encoding
X_train2 = full_pipeline.fit_transform(X_train)
X_val2 = full_pipeline.transform(X_val)

model = getKerasModel(ndim=X_train2.shape[1])
model.fit(X_train2, y_train,epochs=5, batch_size=32, verbose=True, validation_data = (X_val2, y_val))
```
Now this is the result:
```
Train on 265815 samples, validate on 38247 samples
Epoch 1/5
265815/265815 [==============================] - 12s 45us/step - loss: 0.2715 - precision: 0.6690 - recall: 0.4065 - val_loss: 0.2965 - val_precision: 0.6875 - val_recall: 0.4654
Epoch 2/5
265815/265815 [==============================] - 13s 50us/step - loss: 0.2300 - precision: 0.6956 - recall: 0.4959 - val_loss: 0.3735 - val_precision: 0.7000 - val_recall: 0.5192
Epoch 3/5
265815/265815 [==============================] - 13s 47us/step - loss: 0.2111 - precision: 0.7020 - recall: 0.5376 - val_loss: 0.4003 - val_precision: 0.7071 - val_recall: 0.5509
Epoch 4/5
265815/265815 [==============================] - 17s 63us/step - loss: 0.1988 - precision: 0.7113 - recall: 0.5627 - val_loss: 0.4695 - val_precision: 0.7113 - val_recall: 0.5735
Epoch 5/5
265815/265815 [==============================] - 16s 61us/step - loss: 0.1900 - precision: 0.7120 - recall: 0.5831 - val_loss: 0.4893 - val_precision: 0.7134 - val_recall: 0.5908
```
So it definetly says after the last epoch :
```
val_precision: 0.7134 - val_recall: 0.5908
```
When I now use the model, do prediction and evaluate precision and recall seperatly it is much lower:
```
pr = model.predict(X_val2)

binary_result = [1 if i[0] > 0.5 else 0 for i in pr]
print(precision_score(y_val,binary_result))
print(recall_score(y_val,binary_result))

>> 0.23
>> 0.38
```
I don't get why keras is doing much better in its evaluation

**Comment by Saduf2019** [comment]
@tuxmania87  
I ran the code shared by you and face this error as per this [gist](https://colab.sandbox.google.com/gist/Saduf2019/68df26553bf2eb4fb159b09f69f6720d/untitled197.ipynb), please share complete stand alone code or a colab gist to reproduce the issue faced.
